### PR TITLE
eFlora report implemented using JasperReport library and Grails plugin

### PR DESCRIPTION
**Pull request by aruizca**

### CODE REVIEW

File: grails-app/conf/BuildConfig.groovy [hunk 1]
```diff
@@ -47,6 +61,10 @@ grails.project.dependency.resolution = {
         compile ":rendering:1.0.0"
         compile ":csv:0.3.1"
         compile ":wkhtmltopdf:0.1.8"
```
Comment (m-r-c) [inline grails-app/conf/BuildConfig.groovy]: The wkhtmltopdf dependency can be removed now

File: grails-app/services/au/org/ala/profile/hub/ExportService.groovy [hunk 2]
```diff
@@ -126,25 +123,140 @@ class ExportService {
...
-        )
+        curatedModel.options << [
+                allowFineGrainedAttribution: opus.allowFineGrainedAttribution,
+                displayToc: curatedModel.profiles?.size() > 1
+        ]
+
+        curatedModel << [
+                cover: [
+                        title       : opus.title,
+                        subtitle    : curatedModel.profiles[0]?.profile?.fullName,
```
Comment (m-r-c) [inline grails-app/services/au/org/ala/profile/hub/ExportService.groovy]: Might need to test this with a profile that does not have a matched name...I can't remember if the fullName is always populated. If it isn't, then use fullName ?: scientificName.

File: grails-app/services/au/org/ala/profile/hub/ExportService.groovy [hunk 3]
```diff
@@ -126,25 +123,140 @@ class ExportService {
...
+                        logo        : opus.logoUrl,
+                        banner      : opus.bannerUrl,
+                        primaryImage: curatedModel.profiles[0]?.profile?.primaryImage?: (curatedModel.profiles[0]?.profile?.images?.size() > 0 ? curatedModel.profiles[0]?.profile?.images[0].leftImage.largeImageUrl : '')
+                ],
+                colophon: [
+                        copyright: opus.copyrightText,
+                        logo        : opus.logoUrl,
+                        profileLink: "${grailsApplication.config.grails.serverURL}/opus/${params.opusId}/profile/${curatedModel.profiles[0]?.profile?.uuid}"
```
Comment (m-r-c) [inline grails-app/services/au/org/ala/profile/hub/ExportService.groovy]: params.opusId may be the opus shortname, which can be edited. For the PDF, it's probably safer to always use the uuid.

File: grails-app/services/au/org/ala/profile/hub/ExportService.groovy [hunk 4]
```diff
@@ -126,25 +123,140 @@ class ExportService {
...
+     */
+    private Map loadProfileData(String profileId, opus, Map params, boolean latest = false) {
+
+        Map model = [:]
+        model.profile = webService.get("${grailsApplication.config.profile.service.url}/opus/${opus.uuid}/profile/${URLEncoder.encode(profileId, "UTF-8")}?latest=${false}")?.resp
+
+        if (params.taxonomy || params.conservation) {
+            model.profile.speciesProfile = profileService.getSpeciesProfile(model.profile.guid)?.resp
+
+            model.profile.speciesProfile?.conservationStatuses?.each {
```
Comment (m-r-c) [inline grails-app/services/au/org/ala/profile/hub/ExportService.groovy]: We have overlooked the new status list info that Temi added. We'll need to include this in the PDF, too.

File: grails-app/services/au/org/ala/profile/hub/ExportService.groovy [hunk 5]
```diff
@@ -126,25 +123,140 @@ class ExportService {
...
+                        copyright: opus.copyrightText,
+                        logo        : opus.logoUrl,
+                        profileLink: "${grailsApplication.config.grails.serverURL}/opus/${params.opusId}/profile/${curatedModel.profiles[0]?.profile?.uuid}"
+                ]
+        ]
+
+        return curatedModel
+    }
+
+    /**
```
Comment (m-r-c) [inline grails-app/services/au/org/ala/profile/hub/ExportService.groovy]: Remove or fill in empty comments

File: grails-app/services/au/org/ala/profile/hub/ExportService.groovy [hunk 6]
```diff
@@ -104,18 +66,53 @@ class ExportService {
...
+                        'PROFILES_REPORT_OPTIONS': curatedModel.options,
+                        'REPORT_FILE_RESOLVER': new SimpleFileResolver(reportsDir),
+                        'QR_CODE': qrCodeInputStream
+                ]
+        )
+
+        return jasperNonTransactionalService.generateReport(reportDef).toByteArray()
+    }
+
+    /**
```
Comment (m-r-c) [inline grails-app/services/au/org/ala/profile/hub/ExportService.groovy]: Remove or fill in empty comments

**Comment by m-r-c** [comment]
The travis build broke with a dependency error.

File: grails-app/services/au/org/ala/profile/hub/ExportService.groovy (hunk 4 above)
Comment (aruizca) [inline grails-app/services/au/org/ala/profile/hub/ExportService.groovy]: Please clarify

File: grails-app/conf/BuildConfig.groovy (hunk 1 above)
Comment (aruizca) [inline grails-app/conf/BuildConfig.groovy]: Rendering plugin dependency will be removed too

**Comment by aruizca** [comment]
I have completed all the review tasks

**Pull request merged**